examples/CMakeLists.txt

```diff
@@ -60,14 +64,6 @@ if(DNNL_CPU_SYCL)
     endforeach()
 endif()
 
-# Skip examples for CUDA since USM is a default model for the library which is
-# not yet supported for Nvidia backend.
-if(DNNL_SYCL_CUDA)
-    foreach(f ${sources})
-        list(REMOVE_ITEM sources "${f}")
-    endforeach()
-endif()
-
```

Comment:
1. I could reproduce `unimplemented` errors for the following examples:
```bash
gpu-bnorm-u8-via-binary-postops-cpp (Failed)
gpu-primitives-binary-cpp (Failed)
gpu-primitives-layer-normalization-cpp (Failed)
gpu-primitives-lstm-cpp (Failed)
gpu-primitives-prelu-cpp (Failed)
gpu-primitives-reduction-cpp (Failed)
gpu-primitives-reorder-cpp (Failed)
gpu-primitives-shuffle-cpp (Failed)
gpu-rnn-training-f32-cpp (Failed)
gpu-tutorials-matmul-inference-int8-matmul-cpp (Failed)
```
I'd suggest to remove them from the list of source files to build.

2. I can see 312 unimplemented cases for `test_benchdnn_deconv_ci_gpu`. The cases are expected to be implemented therefore the unimplemented status is considered a fail. I think we need to check whether it's an issue or it's correct that they are not implemented. If it's correct then you can skip them in the similar way as it's done [here](https://github.com/oneapi-src/oneDNN/blob/c0c5582d671bf573a576f4e63b2edeb02fbd0f8a/tests/benchdnn/conv/deconv.cpp#L242).
3. I can see an issue with `test_benchdnn_matmul_ci_gpu`. I run `./tests/benchdnn/benchdnn --matmul --engine=gpu --stag=ab --wtag=ab --dtag=ba --runtime_dims_masks=15:15 2048x13:13x512_n"DLRM:0*1"` and get the following output:
```bash
 ** On entry to GEMM_EX  parameter number 16 had an illegal value
terminate called after throwing an instance of 'dnnl::impl::gpu::nvidia::cublas_error'
  what():  At :/onednn/src/gpu/nvidia/cudnn_matmul_impl.hpp : 321cublasGemmEx : CUBLAS_STATUS_INVALID_VALUE
Aborted (core dumped)
```
4. I can see `gpu-sycl-interop-usm-cpp` example fail but I'm not sure why. Do you also see the failure?

Reply:
1. Done
2. The unimplemented cases use s8 which deconv does not support, I have set it to not fail with unimplemented.
3. I have investigated this issue, I believe the problem is with the tests input. This test runs a matrix multiplication of AxB=C. The dimensions appear to be okay however I believe the stride values of C are not correct, though I am not 100% sure.
```
Dimensions A:(2048, 13)
Dimensions B:(13, 512)
Dimensions C:(2048, 512)

Stride A:(13, 1)
Stride B:(512, 1)
Stride C:(1, 2048) -> I think this should be (512, 1)
```
4. I do not see the `gpu-sycl-interop-usm-cpp` fail as long as the `-fsycl-targets=nvptx64-nvidia-cuda` flag is specified when building it.

Reply:
Regarding 4, thanks for the investigation. The stride looks correct to me as we can see from the test that we request a transposed destination which has tag ba (--dtag=ba). Maybe matmil doesn’t handle such cases properly. 

Anyway, this doesn’t seem to be related to adding USM support so we can take care of it later.

Reply:
I have added the `ba` case to unsupported. There are still failing benchmarks when using s8 matrix types with bias, I am still looking into it.

Reply:
Regarding `gpu-sycl-interop-usm-cpp`, I still can see a failure caused by (presumably) an issue with dependencies. When I call `wait()` from the queue the example starts passing, however, when `wait()` is called from the event the example fails.
```
25/147 Test  #25: gpu-sycl-interop-usm-cpp .................................***Failed    4.18 sec
Error in the example: Unexpected output, found a negative value after the ReLU execution..
Example failed on GPU.
```

Reply:
Looking into the issue, I cannot see anything wrong with the cudnn dependency handling. I suspect this may be an issue within dpc++. Would it be possible to change the event wait to a queue wait?

Reply:
Hmm, it seems I didn't rebuild oneDNN properly when I was using the latest DPC++ compiler. I tested it again and the example passes now.

The dependencies work fine. To me, it looks like there was an issue with migrating data between host and device for shared USM. For some reason the migration was not completed by the time we accessed the shared USM on host.